data/txHandler.go

```diff
@@ -172,6 +191,10 @@ func (handler *TxHandler) backlogWorker() {
 		// we have no more post verification items. wait for either backlog queue item or post verification item.
 		select {
 		case wi, ok := <-handler.backlogQueue:
+			if err := wi.capguard.Release(); err != nil {
+				logging.Base().Warnf("Failed to release capacity to ElasticRateLimiter: %v", err)
+			}
+			defer wi.capguard.Served()
```

Comment:
defer? this will queue up until the `backlogWorker` goroutine exit

Reply:
Yep, that's the difference between Served and Release

Release will put capacity back into the ERL, allowing other callers to use the Queue.

Served will notify the congestion manager that the handler has totally "served" a request. This informs the service rate, which is then used for congestion management.

Reply:
my point it this call does not happen until backlogWorker exits. and will build a huge stack of defer calls, is it it?
https://go.dev/play/p/rgEGQc2eV0S

Reply:
oh, I get what you mean now, and yes, that is an issue! Will fix :)